app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -1,30 +1,73 @@
 package com.lightricks.feedexercise.ui.feed
 
+
 import androidx.lifecycle.*
 import com.lightricks.feedexercise.data.FeedItem
+import com.lightricks.feedexercise.network.Constant.BASE_THUMBNAIL_URL
+import com.lightricks.feedexercise.network.FeedApiResponseGenerator
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import com.lightricks.feedexercise.network.TemplatesMetadataItem
 import com.lightricks.feedexercise.util.Event
-import java.lang.IllegalArgumentException
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
 
 /**
  * This view model manages the data for [FeedFragment].
  */
-open class FeedViewModel : ViewModel() {
+open class FeedViewModel () : ViewModel() {
+
+
     private val stateInternal: MutableLiveData<State> = MutableLiveData<State>(DEFAULT_STATE)
     private val networkErrorEvent = MutableLiveData<Event<String>>()
+    private val compositeDisposable = CompositeDisposable()
+
 
+
+    override fun onCleared() {
+        super.onCleared()
+        compositeDisposable.clear()
+    }
     fun getIsLoading(): LiveData<Boolean> {
-        //todo: fix the implementation
-        return MutableLiveData()
+        return stateInternal.map { state -> state.isLoading }
+    }
+
+    private fun fetchFeedItems(){
+        val feedApiService : FeedApiService = FeedApiResponseGenerator.createFeedApiService()
+        val subscribe  = feedApiService.getFeedData()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ feedResponse ->
+                handleResponse(feedResponse)
+            }, { error ->
+                handleNetworkError(error)
+            })
+        compositeDisposable.add(subscribe)
     }
 
+    fun handleNetworkError(error: Throwable){
+        val msg : String = error.message ?: "Unexpected error"
+        networkErrorEvent.postValue(Event(msg))
+    }
+
+    fun handleResponse(feedResponse: TemplatesMetadata) {
+        val rawResponseList = feedResponse.templatesMetadata
+        val feedItemLst : MutableList<FeedItem> = mutableListOf()
+        for(item in rawResponseList){
+            val currItem = FeedItem(item.id, responseUrlToThumbnailUrl(item) ,item.isPremium)
```

Comment:
It can be useful to make a function that converts `TemplatesMetadataItem` to `FeedItem` or 
`List< TemplatesMetadataItem >` to `List< FeedItem >` since you might use it in other places.